### PR TITLE
Patch Rook 1.0.4 Ceph dependency Ceph 14.2.21

### PR DESCRIPTION
**Pull request by murphybytes**
See [Clubhouse Issue](https://app.clubhouse.io/replicated/story/34494/upgrade-ceph-image-to-latest-nautilus-in-rook-1-0-4-addon)

### CODE REVIEW

**Comment by emosbaugh** [comment]
Would you mind adding the diff output of `diff addons/rook/1.0.4 addons/rook/1.0.4-14.2.21`?

Also it looks like upgrades are [failing in testgrid](https://testgrid.kurl.sh/run/pr-1726-b6f6971-rook-1.0.4-14.2.21-k8s-docker?kurlLogsInstanceId=pghrcy#L774). I think we will only have to support upgrades from `1.0.4` to `1.0.4-14.2.21` as we don't currently support upgrades from `1.0.4` to any other version.

**Comment by murphybytes** [comment]
diff --color -r 1.0.4/build-images/ceph/Dockerfile 1.0.4-14.2.21/build-images/ceph/Dockerfile
3c3
< FROM ceph/ceph:v14.2.0
---
> FROM ceph/ceph:v14.2.21
5,32d4
< RUN yum install -y \
<     bind-license \
<     curl \
<     glibc \
<     glibc-common \
<     libcurl \
<     libwbclient \
<     openssl \
<     openssl-libs \
<     perl \
<     perl-Pod-Escapes \
<     perl-libs \
<     perl-macros \
<     python \
<     python-devel \
<     python-libs \
<     python-rtslib \
<     python3 \
<     python3-libs \
<     samba-client-libs \
<     sudo \
<     libldb \
<     nettle \
<     nss \
<     nss-sysinit \
<     nss-tools \
<     openldap \
<   && yum clean all
diff --color -r 1.0.4/build-images/ceph/Makefile 1.0.4-14.2.21/build-images/ceph/Makefile
3c3
< IMAGE := kurlsh/ceph:v14.2.0
---
> IMAGE := kurlsh/ceph:v14.2.21
diff --color -r 1.0.4/build-images/rook-ceph/Dockerfile 1.0.4-14.2.21/build-images/rook-ceph/Dockerfile
5,34c5
< FROM ceph/ceph:v14.2.0
< 
< RUN yum install -y \
<     bind-license \
<     curl \
<     glibc \
<     glibc-common \
<     libcurl \
<     libwbclient \
<     openssl \
<     openssl-libs \
<     perl \
<     perl-Pod-Escapes \
<     perl-libs \
<     perl-macros \
<     python \
<     python-devel \
<     python-libs \
<     python-rtslib \
<     python3 \
<     python3-libs \
<     samba-client-libs \
<     sudo \
<     libldb \
<     nettle \
<     nss \
<     nss-sysinit \
<     nss-tools \
<     openldap \
<   && yum clean all
---
> FROM ceph/ceph:v14.2.21
diff --color -r 1.0.4/build-images/rook-ceph/Makefile 1.0.4-14.2.21/build-images/rook-ceph/Makefile
3c3
< IMAGE := kurlsh/rook-ceph:v1.0.4
---
> IMAGE := kurlsh/rook-ceph:v1.0.4-14.2.21
diff --color -r 1.0.4/install.sh 1.0.4-14.2.21/install.sh
1a2
> 
4c5
<         bail "Rook 1.0.4 is not compatible with Kubernetes 1.20+"
---
>         bail "Rook ${ROOK_VERSION} is not compatible with Kubernetes 1.20+"
27c28
<     kubectl apply -f "$DIR/addons/rook/1.0.4/cluster/object-user.yaml"
---
>     kubectl apply -f "$DIR/addons/rook/${ROOK_VERSION}/cluster/object-user.yaml"
37c38
<     local src="$DIR/addons/rook/1.0.4/operator"
---
>     local src="$DIR/addons/rook/${ROOK_VERSION}/operator"
45c46
<     local src="$DIR/addons/rook/1.0.4/cluster"
---
>     local src="$DIR/addons/rook/${ROOK_VERSION}/cluster"
diff --color -r 1.0.4/operator/ceph-operator.yaml 1.0.4-14.2.21/operator/ceph-operator.yaml
37c37
<         image: kurlsh/rook-ceph:v1.0.4-998b6b6-20210430
---
>         image: kurlsh/rook-ceph:v1.0.4-14.2.21-bf6be4c-20210616

**Comment by murphybytes** [comment]
> Would you mind adding the diff output of `diff addons/rook/1.0.4 addons/rook/1.0.4-14.2.21`?
> 
> Also it looks like upgrades are [failing in testgrid](https://testgrid.kurl.sh/run/pr-1726-b6f6971-rook-1.0.4-14.2.21-k8s-docker?kurlLogsInstanceId=pghrcy#L774). I think we will only have to support upgrades from `1.0.4` to `1.0.4-14.2.21` as we don't currently support upgrades from `1.0.4` to any other version.

```diff
diff --color -r 1.0.4/build-images/ceph/Dockerfile 1.0.4-14.2.21/build-images/ceph/Dockerfile
3c3
< FROM ceph/ceph:v14.2.0
---
> FROM ceph/ceph:v14.2.21
5,32d4
< RUN yum install -y \
<     bind-license \
<     curl \
<     glibc \
<     glibc-common \
<     libcurl \
<     libwbclient \
<     openssl \
<     openssl-libs \
<     perl \
<     perl-Pod-Escapes \
<     perl-libs \
<     perl-macros \
<     python \
<     python-devel \
<     python-libs \
<     python-rtslib \
<     python3 \
<     python3-libs \
<     samba-client-libs \
<     sudo \
<     libldb \
<     nettle \
<     nss \
<     nss-sysinit \
<     nss-tools \
<     openldap \
<   && yum clean all
diff --color -r 1.0.4/build-images/ceph/Makefile 1.0.4-14.2.21/build-images/ceph/Makefile
3c3
< IMAGE := kurlsh/ceph:v14.2.0
---
> IMAGE := kurlsh/ceph:v14.2.21
diff --color -r 1.0.4/build-images/rook-ceph/Dockerfile 1.0.4-14.2.21/build-images/rook-ceph/Dockerfile
5,34c5
< FROM ceph/ceph:v14.2.0
< 
< RUN yum install -y \
<     bind-license \
<     curl \
<     glibc \
<     glibc-common \
<     libcurl \
<     libwbclient \
<     openssl \
<     openssl-libs \
<     perl \
<     perl-Pod-Escapes \
<     perl-libs \
<     perl-macros \
<     python \
<     python-devel \
<     python-libs \
<     python-rtslib \
<     python3 \
<     python3-libs \
<     samba-client-libs \
<     sudo \
<     libldb \
<     nettle \
<     nss \
<     nss-sysinit \
<     nss-tools \
<     openldap \
<   && yum clean all
---
> FROM ceph/ceph:v14.2.21
diff --color -r 1.0.4/build-images/rook-ceph/Makefile 1.0.4-14.2.21/build-images/rook-ceph/Makefile
3c3
< IMAGE := kurlsh/rook-ceph:v1.0.4
---
> IMAGE := kurlsh/rook-ceph:v1.0.4-14.2.21
diff --color -r 1.0.4/install.sh 1.0.4-14.2.21/install.sh
1a2
> 
4c5
<         bail "Rook 1.0.4 is not compatible with Kubernetes 1.20+"
---
>         bail "Rook ${ROOK_VERSION} is not compatible with Kubernetes 1.20+"
27c28
<     kubectl apply -f "$DIR/addons/rook/1.0.4/cluster/object-user.yaml"
---
>     kubectl apply -f "$DIR/addons/rook/${ROOK_VERSION}/cluster/object-user.yaml"
37c38
<     local src="$DIR/addons/rook/1.0.4/operator"
---
>     local src="$DIR/addons/rook/${ROOK_VERSION}/operator"
45c46
<     local src="$DIR/addons/rook/1.0.4/cluster"
---
>     local src="$DIR/addons/rook/${ROOK_VERSION}/cluster"
diff --color -r 1.0.4/operator/ceph-operator.yaml 1.0.4-14.2.21/operator/ceph-operator.yaml
37c37
<         image: kurlsh/rook-ceph:v1.0.4-998b6b6-20210430
---
>         image: kurlsh/rook-ceph:v1.0.4-14.2.21-bf6be4c-20210616

```

**Comment by emosbaugh** [comment]
Upgrade looks good in testgrid. Were you able to manually validate ceph status after an upgrade?

**Comment by murphybytes** [comment]
> Upgrade looks good in testgrid. Were you able to manually validate ceph status after an upgrade?

@emosbaugh yep, looks good.

File: addons/rook/template/testgrid/k8s-docker.yaml [hunk 1]
```diff
@@ -51,3 +51,51 @@
...
+  upgradeSpec:
+    kubernetes:
+      version: "latest"
+    weave:
+      version: "latest"
+    docker: 
+      version: "latest"
+    minio:
+      version: "latest"
+    rook:
```
Comment (emosbaugh) [inline addons/rook/template/testgrid/k8s-docker.yaml:74]: this looks like it was truncated or duplicated or something...

**Review by emosbaugh — CHANGES_REQUESTED** [review]

**Comment by emosbaugh** [comment]
Additionally you will have to add the version to the bottom of this list

https://github.com/replicatedhq/kURL/blob/03f7d70b259792ff990370f1b6112621508662de/web/src/installers/versions.js#L60

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**